docs/kb/integration/defect-dojo-integration.md

```diff
@@ -0,0 +1,114 @@
+---
+tags:
+  - DefectDojo
+  - Integration
+description: How to connect Semgrep and DefectDojo
+---
+
+# How to connect Semgrep and DefectDojo
+
+[DefectDojo](https://www.defectdojo.com/) is a well-known vulnerability management tool. It allows you to gather security issues from tools like Semgrep and more. By integrating Semgrep findings into DefectDojo, security teams gain additional insight into the security posture of their systems.
+
+## Starting with integration
+Integrating Semgrep with DefectDojo means dumping findings detected by Semgrep into DefectDojo.
+There are several steps to follow:
+
+1. In DefectDojo:
+    1 Create your **product**. A [product](https://defectdojo-dev.readthedocs.io/en/latest/features.html#products) is DefectDojo's term for a project.
+    2. In that DefectDojo product, create an **engagement** called `semgrep`. An engagement is a channel to import results.
+2. Run a semgrep scan with flags `--json --output report.json` to generate a JSON report.
+
+Now, there are all the necessary elements to start playing with [DefectDojo API](https://documentation.defectdojo.com/integrations/api-v2-docs/).
+
+### DefectDojo API example 
+
+To run API DefectDojo operations (GET, POST, DELETE, etc..), an API token is necessary. To get it, follow [API guide](https://documentation.defectdojo.com/integrations/api-v2-docs/).
+
+* Once got it in your system, declare the environment variable `DEFECT_DOJO_API_TOKEN`
+```
+export DEFECT_DOJO_API_TOKEN=xxxxxx
+```
+Exploring the DefectDojo API, the endpoint is: `/api/v2/import-scan/` for the first import and `/api/v2/reimport-scan` for the following imports.
+
+These endpoints receive some parameters. Mandatory are:
+* Semgrep report (JSON format)
+* Scan type: `Semgrep JSON Report` for Semgrep importation.
+* Product Name (project name)
+* Engagement Name (a good practice is to name `semgrep`)
```

Comment:
```suggestion
* `engagement_name`: Depending on your use case, the [`engagement_name`](https://defectdojo-dev.readthedocs.io/en/latest/about.html#engagements) can describe a moment in time that the test is taking place in, or it can be used as a simple description about the import. In this case, you can name it `semgrep`.
```

I went through DefectDojo's docs to get a better understanding of how their users might also be using engagement_name.